core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1616,6 +1616,8 @@ class KafkaApis(val requestChannel: RequestChannel,
       val apiVersionRequest = request.body[ApiVersionsRequest]
       if (apiVersionRequest.hasUnsupportedRequestVersion)
         apiVersionRequest.getErrorResponse(requestThrottleMs, Errors.UNSUPPORTED_VERSION.exception)
+      else if (!apiVersionRequest.isValid)
+        apiVersionRequest.getErrorResponse(requestThrottleMs, Errors.INVALID_REQUEST.exception)
```

Comment:
You have a fair point. Would you go with something like `INVALID_NAME_OR_VERSION` for instance? I have thought about this when I wrote the KIP and went with this error. My reasoning was that we may add new fields in the future (e.g. hostname?) so I wanted to have a generic error which is not tight to the client/version cases. `INVALID_REQUEST` sounds quite appropriate for this.

Furthermore, as the client name and version are set by the developer of a client, it should rarely happen when end users use the client library.

Reply:
This should be pretty rare, so it's probably not worth its own error code.  I do wonder if there should be some kind of log message, since otherwise this would be hard for a broker operator to see.

Reply:
I thought about adding a log message as well and I decided to not do it for two reasons:
1) This should be pretty rare and should mainly happen during the development of the client. In this case, the request log could be used; and
2) If a buggy client would be released with an invalid ApiVersionsRequest, I wanted to avoid flooding the logs with something which is not really actionable for a broker operator. What would he do with this information?

Reply:
Those are fair points.  Let's leave it as-is for now.  As you say, the request log will give some hints in this scenario.